# UKPeatlandMicrobes
# Welcome
This repository contains all scripts and data tables used in the analysis of microbial communities from total metagenomes generated from upland peatland soils in the UK, as described in Pallier et al.

The original raw sequence reads are available in the NCBI Sequence Read Archive (SRA) under BioProject accession PRJNA1203648. Prokaryotic metagenome-assembled genomes (MAGs) are accessible through NCBI WGA under the same accession.

# Repository Organization
`Scripts` include `R` scripts used to perform the analyses.

`Tables` contain data tables in `.csv` formats, used in the analyses to produce the data files found in the Data directory. 

`Plots` include the saved figures in `PNG` format.

# Notebook Table of Contents
1. [Map of Study Sites](https://github.com/MalikLab-UOE/UKPeatlandMicrobes/blob/main/Scripts/Figure1a.R)
2. [Ecosystem Properties](https://github.com/MalikLab-UOE/UKPeatlandMicrobes/blob/main/Notebooks/Figure1/Figure1b-i.md)
3. [Microbial Community-level Growth Rates](https://github.com/MalikLab-UOE/UKPeatlandMicrobes/blob/main/Notebooks/Figure2/Figure2.md)
4. [Bioinformatic Workflow](https://github.com/AnantharamanLab/UKPeatlandViruses/blob/main/workflow.ipynb)
5. [Variation in microbial community assembly](https://github.com/MalikLab-UOE/UKPeatlandMicrobes/blob/main/Notebooks/Figure3/Figure3.md)
6. [Microbial community-level metabolic and substrate decomposition functions](https://github.com/MalikLab-UOE/UKPeatlandMicrobes/blob/main/Notebooks/Figure4&5/Figure4-5.md)    
7. [Microbial population-level metabolic potential](https://github.com/MalikLab-UOE/UKPeatlandMicrobes/blob/main/Notebooks/Figure6/Figure6.md)
